From fda8045689cc9feaf6615d1ad20339744726ee12 Mon Sep 17 00:00:00 2001
From: Phil Sutter <psutter@redhat.com>
Date: Thu, 1 Oct 2015 11:46:14 +0200
Subject: [PATCH] ip: handle flush with table > 2^31

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1094676

commit caae16b3b8f7ed609c701d732c03076187780644
Author: Stephen Hemminger <stephen@networkplumber.org>
Date:   Tue Feb 12 11:41:46 2013 -0800

    ip: handle flush with table > 2^31

    Fixes Debian bug #700434
    Need to table id in filter to be unsigned to avoid conversion to -1

    The documentation for "ip" suggests that, when using multiple routing tables, the table ID can be an arbitrary 32 bit number. I've been writing a script that calculates a table Id based on an IP addresses and sets up tables accordingly based on it. This seems to work for everything I've tried except "ip route flush". If you specify a table to flush with an ID over 2^31, it flushes all IPv4 routing tables. For example:

    Will delete all routing tables, including the default one. Needless to say, this is quite annoying. I think this is an upstream bug, but your opinions will be greatly appreciated.

Signed-off-by: Phil Sutter <psutter@redhat.com>
---
 ip/iproute.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ip/iproute.c b/ip/iproute.c
index dbbb1a2..7cbaa16 100644
--- a/ip/iproute.c
+++ b/ip/iproute.c
@@ -85,7 +85,7 @@ static void usage(void)
 
 static struct
 {
-	int tb;
+	unsigned int tb;
 	int cloned;
 	int flushed;
 	char *flushb;
-- 
1.8.3.1

